analyzers/tests/SonarAnalyzer.UnitTest/Helpers/SonarAnalysisContextTest.cs

```diff
@@ -83,7 +106,62 @@ public void ByDefault_ExecuteRule()
                 Verifier.VerifyAnalyzer(testCase.Path,
                                         testCase.Analyzer,
                                         ParseOptionsHelper.FromCSharp8,
-                                        MetadataReferenceFacade.SystemComponentModelPrimitives);
+                                        ConcatReferences(testCase.AdditionalReferences));
+            }
+        }
+
+        [TestMethod]
+        public void WhenProjectType_IsTest_RunsOnlyTestRules()
```

Comment:
Can we improve this name? It currently implies that Main&Test rules are not run. Maybe `WhenProjectType_IsTest_RunRulesWithTestScope`